contracts/ICS20Transfer.sol

```diff
@@ -385,19 +391,22 @@ contract ICS20Transfer is
     {
         IICS20TransferMsgs.FungibleTokenPacketData memory packetData =
             abi.decode(msg_.payload.value, (IICS20TransferMsgs.FungibleTokenPacketData));
-        _refundTokens(msg_.payload.sourcePort, msg_.sourceClient, packetData);
+        address sender = _refundTokens(msg_.payload.sourcePort, msg_.sourceClient, packetData);
+        IBCCallbacksLib.timeoutPacketCallback(sender, msg_);
     }
 
     /// @notice Refund the tokens to the sender
     /// @param sourcePort The source port of the packet
     /// @param sourceClient The source client of the packet
     /// @param packetData The packet data
+    /// @return The address that received the refunded tokens, i.e. sender of the packet
     function _refundTokens(
         string calldata sourcePort,
         string calldata sourceClient,
         IICS20TransferMsgs.FungibleTokenPacketData memory packetData
     )
         private
+        returns (address)
```

Comment:
So, it seems like we're returning the address here to optimize for not having to convert string to address twice.

It feels quite strange, I would prefer to either not optimize this (it would make the `onAcknowledgementPacket` code easier to read as well, or if we think it's important to pass in the sender address to _refundTokens (i.e. convert it before calling in here).
Code readability-wise, I would prefer the non-optimized route.

Reply:
This is true. I was also unsure whether to do this or not. But I do think in the future (or even now if we want to do this for clarity), `_refundTokens` can return other data about the actual refund such as:
- `erc20Address`
- `refundAmount`
- even the `escrowAddress`

I think this looks strange because I only return the sender, but maybe it wouldn't be so strange if I also returned the erc20 address and the refund amount?

Reply:
I see what you mean. It works, but I could imagine that the onAck code could be cleaner if we didn't do this. That said, I'll leave it up to you to decide on this 👍

Reply:
I also returned erc20. Lmk if you like this